src/vaadin-tooltip.html

```diff
@@ -134,25 +144,65 @@
 
           switch (this.position) {
             case 'top':
-              tooltipLeft = targetLeft + horizontalCenterOffset;
               tooltipTop = targetTop - thisRect.height;
+
+              switch (this.align) {
+                case 'left':
+                  tooltipLeft = targetLeft;
+                  break;
+                case 'right':
+                  tooltipLeft = targetRect.left + targetRect.width - thisRect.width;
+                  break;
+                default:
+                  tooltipLeft = targetLeft + horizontalCenterOffset;
+              }
               break;
             case 'bottom':
-              tooltipLeft = targetLeft + horizontalCenterOffset;
               tooltipTop = targetTop + targetRect.height;
+
+              switch (this.align) {
+                case 'left':
+                  tooltipLeft = targetLeft;
+                  break;
+                case 'right':
+                  tooltipLeft = targetRect.left + targetRect.width - thisRect.width;
+                  break;
+                default:
+                  tooltipLeft = targetLeft + horizontalCenterOffset;
+              }
               break;
             case 'left':
               tooltipLeft = targetLeft - thisRect.width;
-              tooltipTop = targetTop + verticalCenterOffset;
+
+              switch (this.align) {
+                case 'top':
+                  tooltipTop = targetTop;
+                  break;
+                case 'bottom':
+                  tooltipTop = targetTop;
+                  break;
+                default:
+                  tooltipTop = targetTop + verticalCenterOffset;
+              }
               break;
             case 'right':
               tooltipLeft = targetLeft + targetRect.width;
-              tooltipTop = targetTop + verticalCenterOffset;
+
+              switch (this.align) {
```

Comment:
The parts of the code seem to be equal, so can be extracted to a function.

Reply:
Done.